### DISCUSSION TITLE
How to get knots to show up in the narrative instead of mph

### DISCUSSION THREAD

**Discussion by sdennstaedt** [weather-gov/api#394] [Q&A]
Would like to see the narrative show up as knots and not mph for a wind forecast.  I know you can change the units on the call such as https://api.weather.gov/gridpoints/LSX/101,90/forecast?units=si.  But that comes up km/h in the forecast narrative and not knots.  At this point it means that I will need to parse the narrative to convert mph to knots unless I'm missing something?

**Comment by StephenClouse** [comment]
The forecast text is meant for the general public, so we would be unlikely to support anything besides mph and km/h there.

The windSpeed property also contains that data, which may be simpler to parse.

                "windSpeed": "10 to 20 mph",

One change we are intending to make soon is to fix windSpeed, along with temperature as well as adding windGust, so instead of strings they are proper [QuantitativeValue](https://schema.org/QuantitativeValue) objects that can be programmatically interpreted without resorting to text parsing.

                "temperature": {
                    "value": "21",
                    "unitCode": "unit:C"
                },
                "windSpeed": {
                    "minValue": "7",
                    "maxValue": "15",
                    "unitCode": "unit:km_h-1"
                },
                "windGust": {
                    "maxValue": "26",
                    "unitCode": "unit:km_h-1"
                },

This is aligned with the rest of the API. We try to present unit+value data in this form always so you can easily get the data to do conversions like you're wanting to do. This is one of the spots that slipped past our review.

**Comment by rasqual** [comment]
The information is more valuable for the huge U.S. maritime community (including the glakes) than for a typical suburban household (who also generally don't care about km/h), it'd be great to see knots.  :-)

**Reply by dwhitemv25** [reply]
In the 5 years since this post was last updated, the QuantitativeValue change was made available as a Feature-Flag. You can set this in the header of your `/gridpoints/.../forecast` query to get the alternate encoding:
```
Feature-Flags: forecast_temperature_qv,forecast_wind_speed_qv
```
